docs/webhook.asciidoc

```diff
@@ -0,0 +1,189 @@
+[id="{p}-webhook"]
+=== Validating webhook
+
+A validating webhook provides additional validation of Elasticsearch resources, it gives you an immediate feedback on the Elasticsearch manifests you submit and allow you to catch errors right away before ECK even tries to fulfill your request.
+
+[float]
+=== Architecture
+The webhook is composed of 4 main components, here is a brief description of each of them to understand how they interact,
+their naming and how they are managed.
+
+. A `ValidatingWebhookConfiguration` object that defines the validating webhook. In ECK it is named `elastic-es-validation.k8s.elastic.co`. It must be created before to start the operator, the `caBundle` field can be automatically managed as part of the
+automatic certificate management _(see below)_.
+. A Kubernetes Service is used to expose the validating server, in ECK its name is fixed to `elastic-webhook-server`.
+It lives in the same Namespace as the webhook server.
+. A webhook server that actually validates the submitted resources. In ECK it is the operator itself when it
+is configured to have the `webhook` role. See <<{p}-operator-config,Configuring ECK>> if you want to learn more about the
+`operator-roles` flag.
+. A Secret which contains the required certificates to secure the connection between the API server and the webhook.
+Like the ValidatingWebhookConfiguration it must be created before starting the operator, even if it is empty.
+If ECK has been installed with the provided manifest its name is `elastic-webhook-server-cert`.
+The content of this Secret and the lifecycle of the certificates are automatically managed for you. You can disable
+this feature if you want to setup the certificates by yourself or with the cert-manager, see an example for the latter below.
+
+[float]
+=== Managing webhook certificate with cert-manager
+
+If the operator is currently running you first need to ensure that the automatic certificate management is disabled.
+This can be done by updating the operator deployment manifest and adding the `--manage-webhook-certs=false` flag.
+
+Then, if it is not already done you have to install cert-manager v0.11+ as described in the https://docs.cert-manager.io/en/latest/getting-started/install/[cert-manager documentation].
+
+Below is a complete example to create all of the resources necessary for the validating webhook to function.
```

Comment:
The following example shows how to create all the resources that a webhook requires to function.